### PR TITLE
Update EIP-1: Clarify when to use `requires`

### PR DESCRIPTION
**Pull request by SamWilsn**
As discussed in [EIPIP Meeting #64](https://github.com/ethereum-cat-herders/EIPIP/issues/174), here's a PR with my proposal for updating the `requires` text.

### CODE REVIEW

File: EIPS/eip-1.md
Comment (Pandapip1) [inline EIPS/eip-1.md:188]:
```suggestion
A `requires` dependency is created when the current EIP cannot be understood or implemented without a concept or technical element from another EIP. Mentioning another EIP within the Specification section automatically creates a dependency. Mentioning another EIP outside of the Specification section does not necessarily create a dependency.
```
My personal preference is this.

Comment (xinbenlv) [reply EIPS/eip-1.md:188]: Good. Happy that my poor English helps.

Comment (xinbenlv) [reply EIPS/eip-1.md:188]: I support this principle, so long as we also require authors keep Specification as minimum as possible: Leave anything that's rationale, motivation or reference implementations _out of_ Specification section.

Comment (SamWilsn) [reply EIPS/eip-1.md:188]: EIP-1 doesn't really use RFC 2119 (and related), so I don't really like using the keywords in all caps.

Comment (xinbenlv) [reply EIPS/eip-1.md:188]: Just curious, what was our rationale that EIP-1 didn't want to adopt RFC 2119? 
I don't have strong opinion to preserve the cap, but I weakly prefer having a cap make it more obvious it is a rule rather than just a description.

**Comment by SamWilsn** [comment]
Personally, I wouldn't explicitly mention a section at all, but I do see the benefit of a hard-and-fast rule.

I think I prefer @Pandapip1's edit over @xinbenlv's because the crux of the decision isn't _where_ the reference occurs, but _why_ you're referencing it.

For example, if an EIP uses `CREATE2`, you can pretty easily write the entire EIP without explicitly mentioning EIP-1014 in the specification section&mdash;the opcode is well-known enough. That shouldn't mean you don't have to put `requires: 1014` in the preamble.

File: EIPS/eip-1.md
Comment (Pandapip1) [reply EIPS/eip-1.md:188]: > Just curious, what was our rationale that EIP-1 didn't want to adopt RFC 2119?

Readability, most likely. I definitely would support adopting RFC 2119 for EIP-1 though.

**Comment by xinbenlv** [comment]
> Personally, I wouldn't explicitly mention a section at all, but I do see the benefit of a hard-and-fast rule.
> 
> I think I prefer @Pandapip1's edit over @xinbenlv's because the crux of the decision isn't _where_ the reference occurs, but _why_ you're referencing it.
> 
> For example, if an EIP uses `CREATE2`, you can pretty easily write the entire EIP without explicitly mentioning EIP-1014 in the specification section—the opcode is well-known enough. That shouldn't mean you don't have to put `requires: 1014` in the preamble.

I agree 100% of what you say: what we really care is the dependency between EIPs. 

but I have to admit it's a bit hard to judge, sometimes also subjective. For example, EIP-155 introduces ChainID, which https://eips.ethereum.org/EIPS/eip-2718 depends on. But it didn't list 155 as required. Also ChainId is needed for many hard-fork meta EIPs, but they don't necessarily consider 155. This principle makes it harder to judge. Therefore that's why I slightly tweak the wording so that if author feel it's necessary for spec to reference that EIP, they will put it in requires, otherwise no mandate.

**Comment by Pandapip1** [comment]
> For example, EIP-155 introduces ChainID, which https://eips.ethereum.org/EIPS/eip-2718 depends on. But 2718 didn't list 155 as required.

Oops, that's a mistake. I'll go fix that for them :)

**Comment by Pandapip1** [comment]
> Also ChainId is needed for many hard-fork meta EIPs, but they don't necessarily consider 155 as a dependency

More edits I guess :P

**Comment by Pandapip1** [comment]
> but I have to admit it's a bit hard to judge, sometimes also subjective. For example, EIP-155 introduces ChainID, which https://eips.ethereum.org/EIPS/eip-2718 depends on. But 2718 didn't list 155 as required.

I don't see chainid mentioned in the specification?

**Comment by xinbenlv** [comment]
> > but I have to admit it's a bit hard to judge, sometimes also subjective. For example, EIP-155 introduces ChainID, which https://eips.ethereum.org/EIPS/eip-2718 depends on. But 2718 didn't list 155 as required.
> 
> I don't see chainid mentioned in the specification?

https://eips.ethereum.org/EIPS/eip-2718 Specification > Receipts
> LegacyTransaction is rlp([nonce, gasPrice, gasLimit, to, value, data, v, r, s])

In which `v` has the chainId. 
See, it just prove my point this is so unobvious.

**Comment by gcolvin** [comment]
For Core stuff it's really difficult to track things back to their original EIP once they are into the protocol.  And once they are into the Yellow Paper I think they can pretty much be taken for granted.  The Requires section seems essential mostly for the other, related EIPs that are also still in-process.  For ERCs that's not quite so hard a problem - the ERC really is the controlling technical document, and they don't tend to keep building on top of each other as much as Core EIPs do.

File: EIPS/eip-1.md
Comment (lightclient) [reply EIPS/eip-1.md:188]: I think the existing text is fine. We don't need to spoon feed authors.

**Review by lightclient — APPROVED** [review]

File: EIPS/eip-1.md
Comment (Pandapip1) [reply EIPS/eip-1.md:188]: We need to spoon-feed bots, though. @SamWilsn can decide whether this wording change is valuable or not.

Comment (Pandapip1) [reply EIPS/eip-1.md:188]: > Probably out of scope for this PR though.

~~probably~~ definitely

Comment (SamWilsn) [reply EIPS/eip-1.md:188]: We can keep the general text that @lightclient prefers, and create a rule in `eipw` to enforce @Pandapip1's rule.

**Review by Pandapip1 — APPROVED** [review]

**Review by xinbenlv — APPROVED** [review]
This wording seems to me in good state to merge, are we waiting for anything else before merge it? @SamWilsn

**Pull request merged**
